handler/OnSettingUpdatedHandler.ts

```diff
@@ -10,23 +10,25 @@ export class OnSettingUpdatedHandler {
     constructor(private readonly app: IApp, private readonly read: IRead, private readonly http: IHttp) {}
 
     public async run() {
-        const botId = await getAppSettingValue(this.read, AppSetting.DialogflowBotId);
-        const clientEmails = (await getAppSettingValue(this.read, AppSetting.DialogflowClientEmail)).split(',');
-        const privateKeys = (await getAppSettingValue(this.read, AppSetting.DialogFlowPrivateKey)).split(',');
-        const privateKey = privateKeys.length >= botId ? privateKeys[botId - 1] : privateKeys[0];
-        const clientEmail = clientEmails.length >= botId ? clientEmails[botId - 1] : clientEmails[0];
+        const dialogflowBotList = JSON.parse(await getAppSettingValue(this.read, AppSetting.DialogflowBotList));
```

Comment:
`JSON.parse()` can through parsing error if the config is invalid JSON. Can we handle that error and add proper error log for it?

Reply:
@Shailesh351 Added Handlers for invalid JSON and undefined omnichannel agent username